### PR TITLE
Fix PDF download, updated for new Payment API / SBC Pay API

### PR DESCRIPTION
**Pull request by bcdevlucas**
*Issue #:* /bcgov/entity###
/bcgov/entity#5318

*Description of changes:*

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the namerequest license (Apache 2.0).

### CODE REVIEW

File: client/src/components/payment/payment-mixin.ts [hunk 1]
```diff
@@ -17,6 +17,10 @@ export default class PaymentMixin extends Vue {
     return this.$store.getters[paymentTypes.GET_SBC_PAYMENT]
   }
 
+  get sbcPaymentStatus () {
+    return this.$store.getters[paymentTypes.GET_SBC_PAYMENT_STATUS]
```
Comment (severinbeauvais) [inline client/src/components/payment/payment-mixin.ts:21]: At some point, "paymentTypes" should be changed to PascalCase so it's not confused with a property (which should be camelCase).

File: client/src/modules/payment/models.ts [hunk 2]
```diff
@@ -1,5 +1,10 @@
 import { ErrorI } from '@/modules/error/store/actions'
 
+export const PaymentStatus = {
```
Comment (severinbeauvais) [inline client/src/modules/payment/models.ts:3]: If this type is the same as SbcPaymentStatus, why have both?

File: client/src/modules/payment/store/getters.ts [hunk 3]
```diff
@@ -10,6 +10,9 @@ export default {
   [types.PAYMENT_COMPLETE_MODAL_IS_VISIBLE]: state => state[STATE_KEY].isPaymentCompleteVisible,
   [types.PAYMENT_IS_IN_PROGRESS]: state => state[STATE_KEY].isProcessing,
   [types.GET_SBC_PAYMENT]: state => state[STATE_KEY].sbcPayment,
+  [types.GET_SBC_PAYMENT_STATUS]: state => (state[STATE_KEY].sbcPayment)
```
Comment (severinbeauvais) [inline client/src/modules/payment/store/getters.ts:13]: "types" should be capitalized to distinguish it from a property/variable.

File: client/src/modules/payment/models.ts (hunk 2 above)
Comment (bcdevlucas) [reply client/src/modules/payment/models.ts:3]: The codes are different on both sides for various reasons including that a payment record on their end is not 100% analagous to a payment on ours. Our payment only references thee SbcPayment. If we need the SbcPayment we do not get it from our side, we always make the API call as we cannot control what happens on the SbcPayment side of things.

File: client/src/modules/payment/store/getters.ts (hunk 3 above)
Comment (bcdevlucas) [reply client/src/modules/payment/store/getters.ts:13]: Not enforced by eslint, and those are not actually 'types' referenced in those files, they're exported string constants so technically they are just exported variables. I could have written 'action_dictionary.<type>' or 'all_actions.<type> I guess but yeah...

File: client/src/components/payment/payment-mixin.ts (hunk 1 above)
Comment (bcdevlucas) [reply client/src/components/payment/payment-mixin.ts:21]: Not enforced by eslint, and those are not actually 'types' referenced in those files, they're exported string constants so technically they are just exported variables. I could have written 'action_dictionary.<type>' or 'all_actions.<type> I guess but yeah...

File: client/src/modules/payment/models.ts (hunk 2 above)
Comment (severinbeauvais) [reply client/src/modules/payment/models.ts:3]: I agree with that approach. The 2 types just look very similar to me :)

File: client/src/components/payment/payment-mixin.ts (hunk 1 above)
Comment (severinbeauvais) [reply client/src/components/payment/payment-mixin.ts:21]: You're right, but they're used as types/enum.

Please fix if you're in this area again.

**Pull request merged**